tests/app/job/test_rest.py

```diff
@@ -762,3 +763,89 @@ def test_get_all_notifications_for_job_returns_csv_format(
+
+    service_id = sample_template.service.id
+
+    path = '/service/{}/job/job-stats'.format(service_id)
+    auth_header = create_authorization_header()
+    response = client.get(path, headers=[auth_header])
+    assert response.status_code == 200
+    resp_json = json.loads(response.get_data(as_text=True))
+    assert len(resp_json['data']) == 5
+    assert resp_json['data'][0]["job_id"]
+    assert resp_json['data'][0]["created_at"]
+    assert not resp_json['data'][0]["scheduled_for"]
+    assert resp_json['data'][0]["template_id"]
+    assert resp_json['data'][0]["template_version"]
+    assert resp_json['data'][0]["service_id"]
+    assert resp_json['data'][0]["requested"]
+    assert resp_json['data'][0]["sent"] == 0
+    assert resp_json['data'][0]["delivered"] == 0
+    assert resp_json['data'][0]["failed"] == 0
+
+
+def test_get_jobs_raises_for_bad_limit_days(client, sample_service):
+    path = '/service/{}/job/job-stats'.format(sample_service.id)
+    auth_header = create_authorization_header()
+    response = client.get(path,
+                          query_string={'limit_days': 'bad_number'},
+                          headers=[auth_header])
+    assert response.status_code == 400
+    assert response.get_data(as_text=True) == '{\n  "message": {\n    "limit_days": [\n      ' \
+                                              '"bad_number is not an integer"\n    ]\n  },\n  "result": "error"\n}'
+
+
+def test_parse_status_turns_comma_sep_strings_into_list():
+    statuses = "started, finished, pending"
+    from app.job.rest import _parse_statuses
+    assert _parse_statuses(statuses) == ["started", "finished", "pending"]
+
+
+def test_parse_status_turns_empty_string_into_empty_list():
+    statuses = ""
+    from app.job.rest import _parse_statuses
+    assert _parse_statuses(statuses) == ['']
+
+
+def test_get_job_stats_by_service_id_and_job_id(client, sample_job):
+    auth_header = create_authorization_header()
+    response = client.get("/service/{}/job/job-stats/{}".format(sample_job.service_id, sample_job.id),
+                          headers=[auth_header])
+    assert response.status_code == 200
+    resp_json = json.loads(response.get_data(as_text=True))
+    assert resp_json["job_id"] == str(sample_job.id)
+    assert resp_json["created_at"] == sample_job.created_at.strftime(DATETIME_FORMAT)
+    assert not resp_json["scheduled_for"]
+    assert resp_json["template_id"] == str(sample_job.template_id)
+    assert resp_json["template_version"] == sample_job.template_version
+    assert resp_json["service_id"] == str(sample_job.service_id)
+    assert resp_json["requested"] == sample_job.notification_count
+    assert resp_json["sent"] == 0
+    assert resp_json["delivered"] == 0
+    assert resp_json["failed"] == 0
+
+
+def test_get_job_stats_with_invalid_job_id_returns404(client, sample_template):
+    service_id = sample_template.service.id
+    path = '/service/{}/job/job-=stats{}'.format(service_id, "bad-id")
```

Comment:
think there's a typo in this bad id (unless you intended this)

should it not still be `/service/{}/job/job-stats/{}`

cos this will 404 because of the routing engine not recognising the path, rather than the job id not matching anything in the DB which is what we should be testing